modules/store/src/action_creator.ts

```diff
@@ -0,0 +1,47 @@
+// ==========
+// = Adapted from: rex-tils
+// = https://github.com/Hotell/rex-tils
+// ==========
+
+/**
+ * ActionWithProps can exist in only 2 shapes; type only or type with properties
+ */
+type ActionWithProps<T extends string = string, P = void> = P extends void
+  ? Readonly<{ type: T }>
+  : Readonly<{
+      type: T;
+    }> &
+      Readonly<P>;
+
+export function createAction<T extends string>(type: T): ActionWithProps<T>;
+export function createAction<
+  T extends string,
+  P extends {
+    [key: string]: any;
+  }
+>(type: T, payload: P): ActionWithProps<T, P>;
+export function createAction<
+  T extends string,
+  P extends {
+    [key: string]: any;
+  }
+>(type: T, payload?: P) {
+  // Spread types may only be created from object types. ts(2698) Typescript 3.1.1

```

Comment:
Since we're on TS 3.1.1, I think we can use this line? 😄

Reply:
This requires TS 3.2. :smile:
[Generic spread expressions in object literals](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-2.html)

Reply:
👍 	Got it.

Will be something for v8 then 😄